envoy/http/alternate_protocols_cache.h

```diff
@@ -128,9 +144,17 @@ class AlternateProtocolsCache {
    * @return the number if entries in the map.
    */
   virtual size_t size() const PURE;
+
+  /**
+   * @param origin The origin to get HTTP/3 status for.
+   * @return the status tracker if there is one. Otherwise, nullptr.
+   */
+  virtual std::shared_ptr<AlternateProtocolsCache::Http3StatusTracker>
```

Comment:
I'm mildly surprised that this returns a shared_ptr. Could it return a ref instead? (Presumably not or you would likely have done that :>)

Reply:
The life time of an entry in this cache may short live the conn pool. But the pool may want to modify the status tracker while the entry is already removed from cache.